src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritItemListenerTest.java

```diff
@@ -92,4 +101,38 @@ public void testOnLoadedWithNoConnection() throws Exception {
         listener.onLoaded();
         Mockito.verify(gerritServer, Mockito.times(0)).startConnection();
     }
+
+    /**
+     * Test {@link GerritItemListener#onLocationChanged(hudson.model.Item, String, String)} do handle renamed job.
+     *
+     * @throws Exception if so.
+     */
+    @Test
+    @Issue("JENKINS-27651")
+    public void testOnJobRenamed() throws Exception {
+        FreeStyleProject job = j.createFreeStyleProject("MyJob");
+        PatchsetCreated event = Setup.createManualPatchsetCreated();
+
+        GerritTrigger trigger = spy(new GerritTrigger(
+                null, null, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                true, true, true, false, false, "", "", "", "", "", "", "", null, null, null,
+                null, false, false, "", null));
+
+        doReturn(new GerritTrigger.DescriptorImpl()).when(trigger, "getDescriptor");
+        job.addTrigger(trigger);
+        trigger.start(job, true);
+
+        when(trigger.isInteresting(event)).thenReturn(true);
+        job.renameTo("MyJobRenamed");
+        GerritItemListener listener = new GerritItemListener();
+        listener.onLocationChanged(job, "MyJob", "MyJobRenamed");
+        PluginImpl.getInstance().getHandler().notifyListeners(event);
+        Jenkins.getInstance().getQueue().maintain();
+        Thread.sleep(DELAY);
+        Assert.assertNotNull(job.getLastBuild());
+    }
+
+    /** some delay to ensure the job is ran after being scheduled. */
+    public static final int DELAY = 100;
```

Comment:
Checkstyle complains about any numeric literals above ```1``` But in test code its OK to add 

```//CS IGNORE MagicNumber FOR NEXT 100 LINES. REASON: Test code```

Simpler than to have a separate set of checkstyle rules for test code and production.